Parsers/ASimRegistryEvent/Parsers/vimRegistryEventMicrosoft365D.yaml

```diff
@@ -1,108 +1,156 @@
-
-                  // Processes
-                | extend
-                  ActingProcessId = tostring(InitiatingProcessId), 
-                  ParentProcessId = tostring(InitiatingProcessParentId) 
-                | project-away InitiatingProcessId, InitiatingProcessParentId
-
-                | project-rename
-                  ParentProcessName = InitiatingProcessParentFileName, 
-
-                  ParentProcessCreationTime = InitiatingProcessParentCreationTime, 
-
-                  ActingProcessName = InitiatingProcessFolderPath, 
-                  ActingProcessFileName = InitiatingProcessFileName,
-                  ActingProcessCommandLine = InitiatingProcessCommandLine, 
-
-                  ActingProcessMD5 = InitiatingProcessMD5, 
-                  ActingProcessSHA1 = InitiatingProcessSHA1, //OK
-                  ActingProcessSHA256 = InitiatingProcessSHA256, 
-                  ActingProcessIntegrityLevel = InitiatingProcessIntegrityLevel, 
-                  ActingProcessTokenElevation = InitiatingProcessTokenElevation, 
-                  ActingProcessCreationTime = InitiatingProcessCreationTime 
-
-                  // -- aliases
-                | extend 
-                  Username = ActorUsername,
-                  UserId = ActorUserId,
-                  UserIdType = ActorUserIdType,
-                  User = ActorUsername,
-                  CommandLine = ActingProcessCommandLine,
-                  Process = ActingProcessName
-
-
-               };
-               RegistryEvents_M365D
\ No newline at end of file
+  let RegistryType = datatable (TypeCode: string, TypeName: string)
+    [
+    "None", "Reg_None",
+    "String", "Reg_Sz",
+    "ExpandString", "Reg_Expand_Sz",
+    "Binary", "Reg_Binary",
+    "Dword", "Reg_DWord",
+    "MultiString", "Reg_Multi_Sz",
+    "QWord", "Reg_QWord"
+  ];
+  let parser = (
+      starttime: datetime=datetime(null), 
+      endtime: datetime=datetime(null),
+      eventtype_in: dynamic=dynamic([]),
+      actorusername_has_any: dynamic=dynamic([]),
+      registrykey_has_any: dynamic =dynamic([]),
+      registryvalue_has_any: dynamic =dynamic([]),
+      registrydata_has_any: dynamic =dynamic([]),
+      dvchostname_has_any: dynamic=dynamic([]),
+      disabled: bool=false
+      ) {
+      DeviceRegistryEvents
+      | where not(disabled)
+      | where (isnull(starttime) or TimeGenerated >= starttime) 
+          and (isnull(endtime) or TimeGenerated <= endtime)
+      | where (array_length(eventtype_in) == 0 or ActionType in~ (eventtype_in)) and
+          (array_length(actorusername_has_any) == 0 or (InitiatingProcessAccountName has_any (actorusername_has_any)) or (InitiatingProcessAccountDomain has_any (actorusername_has_any)) or (strcat(InitiatingProcessAccountDomain, '\\', InitiatingProcessAccountName) has_any (actorusername_has_any))) and
+          (array_length(registrykey_has_any) == 0 or RegistryKey has_any (registrykey_has_any)) and 
```

Comment:
thanks Anki. Yes, this was needed. updated

Reply:
PrevRegistryKey need to be considered as RegistryKey only when ActionType in ("RegistryKeyDeleted", "RegistryValueDeleted") as per extend statement. But During the filter its always considered irrespective of this condition. Can you please clarify?

Reply:
@anki-narravula I included both RegistryKey and PreviousRegistryKey in filter because during 'Delete' events, RegistryKey and RegistryValueName will come as null